### PR TITLE
test: Increase Email test coverage and enable component in random tests

### PR DESCRIPTION
**Pull request by gr8man**
**Description**
Ref: #9968

- Added additional test cases for the `Email` class (covering header injection prevention, SMTP disconnection handling, UTF-8 filenames, and other edge cases).
- Resolved global state pollution issues in namespace overrides by implementing cleanup in `tearDown()`.
- Enabled the `Email` component in the random tests execution configuration (`random-tests-config.txt`).
- Fixed a PHPDoc type typo for the `$newline` property in `Email.php`.

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value (without duplication)
- [x] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Email/Email.php [hunk 1]
```diff
@@ -197,7 +197,7 @@ class Email
      *
      * @see http://www.ietf.org/rfc/rfc822.txt
      *
-     * @var "\n"|"\r\n"
+     * @var string
```
Comment (michalsn) [inline system/Email/Email.php]: You're weakening annotations. Please revert.

File: tests/system/Email/EmailTest.php [hunk 2]
```diff
@@ -31,6 +31,37 @@ final class EmailTest extends CIUnitTestCase
 {
     use ReflectionHelper;
 
+    protected function tearDown(): void
```
Comment (michalsn) [inline tests/system/Email/EmailTest.php]: If this method is needed, you have to call `parent::tearDown()`.

**Review by michalsn — CHANGES_REQUESTED** [review]
Please remove the newly added tests that depend on the namespace-level native-function mocks, together with the global variables used to control them. 

Although namespace-level function shadowing can be useful for isolated cases, using it here across nine native functions creates persistent process-level behavior and substantial global test state. This makes the tests difficult to isolate and maintain, and several mocks replace native behavior even when a test has not explicitly configured them.

**Comment by gr8man** [comment]
I cannot fix the PHPDoc type typo (@var "\r\n"|"n") directly in system/Email/Email.php because of an upstream issue with PHP-CS-Fixer's global_namespace_import rule.

Since Email.php has a namespace (CodeIgniter\Email), the fixer incorrectly treats the \n inside the PHPDoc string as a global class, stripping the backslash and erroneously injecting a use n; statement at the top of the file, which breaks the build.

File: tests/system/Email/EmailTest.php (hunk 2 above)
Comment (michalsn) [reply tests/system/Email/EmailTest.php]: Please remove the entire method if nothing is happening in it.

File: tests/system/Email/EmailTest.php [hunk 3]
```diff
@@ -409,4 +415,486 @@ public function testPrepQuotedPrintableUnwrapRemovesTagsOnly(): void
...
+        $this->assertStringContainsString('Content-Type: multipart/mixed', (string) $headerStr);
+        $this->assertStringContainsString('Content-Type: image/png', (string) $finalBody);
+        $this->assertStringContainsString('Content-Disposition: attachment', (string) $finalBody);
+    }
+
+    public function testWordWrap(): void
+    {
+        $email = new Email();
+        /** @phpstan-ignore assign.propertyType */
+        $email->newline = "\n";
```
Comment (michalsn) [inline tests/system/Email/EmailTest.php]: Please use the public API instead - we have `setNewline()`.

File: tests/system/Email/EmailTest.php [hunk 4]
```diff
@@ -409,4 +415,486 @@ public function testPrepQuotedPrintableUnwrapRemovesTagsOnly(): void
...
+        $email = new Email();
+        $this->assertFalse($email->attach('/path/to/nonexistent/file.txt'));
+    }
+
+    public function testAttachUnreadableFile(): void
+    {
+        $email = new Email();
+        // A directory is not readable as a regular file for fopen
+        $this->assertFalse($email->attach(__DIR__));
+    }
```
Comment (michalsn) [inline tests/system/Email/EmailTest.php]: This does not test an unreadable file. `__DIR__` fails the earlier `is_file()` check - same as in previous test. Please remove.

**Review by michalsn — APPROVED** [review]
Thank you!

**Review by paulbalandan — APPROVED** [review]

**Review by ni441 — APPROVED** [review]

**Comment by michalsn** [comment]
Thank you @gr8man
